### PR TITLE
Add Support for Swift 3

### PR DESCRIPTION
**Pull request by RafaelPlantard**
- Refactor Package for Swift 3;
- Add unit tests to ensure the correct operation at the package.
#18

### CODE REVIEW

File: Randoms.swift [hunk 1]
```diff
@@ -19,9 +19,18 @@ public extension Bool {
 public extension Int {
     /// SwiftRandom extension
     public static func random(_ range: Range<Int>) -> Int {
-        return random(range.upperBound, range.lowerBound)
+		#if swift(>=3)
+        return random(range.lowerBound, range.upperBound - 1)
+		#else
+		return random(range.upperBound, range.lowerBound)
+		#endif
```
Comment (thellimist) [inline Randoms.swift]: Please use only tabs or only spaces. SwiftRandom uses 4 spaces indentation. General practice is using the same indentation the project was already using. XCode has an option to convert all tabs to spaces and vice-versa.

File: Randoms.swift [hunk 2]
```diff
@@ -125,8 +134,13 @@ public extension ArraySlice {
...
             return nil
         }
-        
-        let index = Int.random(self.startIndex, self.endIndex)
+
+		#if swift(>=3)
+        let index = Int.random(self.startIndex, self.count - 1)
+		#else
+		let index = Int.random(self.startIndex, self.endIndex)
+		#endif
```
Comment (thellimist) [inline Randoms.swift]: same here

Comment (thellimist) [inline Randoms.swift]: If you think it's necessary. Just editing xcode style or opening in a text editor like sublime should be sufficient but It's up to you.

**Comment by RafaelPlantard** [comment]
@thellimist I have used Swimat and Structure > Re-indent to ensure the correct indentation.
screen shot 2016-10-04 at 10 00 07

**Comment by thellimist** [comment]
Looks good. I'll test when I've access to XCode.

**Comment by RafaelPlantard** [comment]
Thanks @thellimist

**Comment by thellimist** [comment]
@RafaelPlantard My xcode gives tons of errors. Does this work without error in your build?

**Comment by RafaelPlantard** [comment]
Yah: Xcode8:

screen shot 2016-10-06 at 18 17 25

**Comment by RafaelPlantard** [comment]
I update my fork with your repository to be sure:
screen shot 2016-10-06 at 18 19 05

**Comment by thellimist** [comment]
My fault. It's working perfect. NIce job!

**Pull request merged**